core/src/main/java/org/kiwix/kiwixmobile/core/downloader/ChunkUtils.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.downloader
+
+import org.kiwix.kiwixmobile.core.utils.StorageUtils.getFileNameFromUrl
+
+object ChunkUtils {
+  private const val ALPHABET = "abcdefghijklmnopqrstuvwxyz"
+  private const val ALPHABET_SIZE = 26
+  private const val ZIM_EXTENSION = ".zim"
+  const val PART = ".part.part"
+  const val CHUNK_SIZE = 1024L * 1024L * 1024L * 2L
+
+  fun getChunks(
+    url: String?,
+    contentLength: Long,
+    notificationID: Int
+  ): List<Chunk> {
+    val fileCount = getZimChunkFileCount(contentLength)
+    val filename = getFileNameFromUrl(url)
+    val fileNames = getZimChunkFileNames(filename, fileCount)
+    return generateChunks(contentLength, url, fileNames, notificationID)
+  }
+
+  private fun generateChunks(
+    contentLength: Long,
+    url: String?,
+    fileNames: Array<String?>,
+    notificationID: Int
+  ): List<Chunk> {
+    val ranges = fileNames.indices.map { it * (CHUNK_SIZE + 1) }.map { it..it + CHUNK_SIZE }
+    val filenamesAndRanges: List<Pair<String?, LongRange>> = fileNames.zip(ranges)
+    val chunks: MutableList<Chunk> = ArrayList()
```

Comment:
You don't need this variable or to use `mapTo`, returning a `map` is all you need to do.

I'd also do away with the varaible `filenamesAndRanges` as `fileNames.zip(ranges)` is well understood enough and the destructured lambda params aid even more

Reply:
But without using mapTo() how could I return chunk every time. I mean, please correct me if I am wrong, using a map transformation, we get the same list transformed.  But here we want to create new chunks for each item in fileAndRanges using fileNames and current range. After filename.zip(ranges) returned object will be of List<Pair>, how will the object get converted into List<Chunk> without using a new variable chunks( which is in this case)

Reply:
`.map` transforms the type, like how you turned indices into ranges

`.map` is just a much nicer way of saying

```
val newList = mutableListOf()
for(item in someOtherList){
newList.add(NewItem(item))
}
return newList
```

return the map

Reply:
@macgills Changes made. Please review